core/types.go

```diff
@@ -380,6 +382,14 @@ type SignedData interface {
 	json.Marshaler
 }
 
+type Eth2SignedData interface {
+	SignedData
+	// DutyType returns Duty type associated with underlying signed data.
+	DutyType() DutyType
```

Comment:
Rather do `DomainName()` as discussed, since domain is actually a eth2 thing while duty type is not a eth2 thing.